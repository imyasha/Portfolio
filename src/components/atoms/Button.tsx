import React from 'react';
import styled from 'styled-components';

const Button = (props: any) => {
  const { children, href, rel, target } = props;

  return (
    <StyledButton
      href={href}
      target={target || undefined}
      rel={rel || target === '_blank' ? 'noopener noreferrer' : undefined}
      {...props}
    >
      {children}
    </StyledButton>
  );
};

export default Button;

const StyledButton = styled.a`
  background: #fff;
  border: 2px solid #e94e1b;
  color: #e94e1b;
  padding: 0.5rem 2rem;
  border-radius: 99px;
  margin: 1rem 0;
  transition: 0.2s;
  display: inline-block;
  text-decoration: none;
  &:hover {
    background: #e94e1b;
    color: #fff;
  }
`;
