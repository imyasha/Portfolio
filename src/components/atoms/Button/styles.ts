import styled from 'styled-components';

const StyledButton = styled.button`
	display: inline-block;
	margin: 2rem auto;
	border: 2px solid ${({ theme }) => theme.text};
	color: ${({ theme }) => theme.text};
	padding: 0.5rem 3rem;
	border-radius: 5px;
	background: transparent;
`;

const StyledLink = styled.a`
	display: inline-block;
	margin: 2rem auto;
	border: 2px solid ${({ theme }) => theme.text};
	color: ${({ theme }) => theme.text};
	padding: 0.5rem 3rem;
	border-radius: 5px;
	background: transparent;
	text-decoration: none;

	&:hover {
		border: 2px solid ${({ theme }) => theme.headline};
		color: ${({ theme }) => theme.headline};
	}
`;

export { StyledButton, StyledLink };
