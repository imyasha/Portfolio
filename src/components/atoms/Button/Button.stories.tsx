import { Button, ButtonProps } from './Button';

export default {
	title: 'atoms/Button',
	component: Button,
};

export const ButtonStory = (args: ButtonProps) => <Button {...args} />;

ButtonStory.storyName = 'Button';
ButtonStory.args = {
	children: 'Button Text',
	href: '/link',
	target: '_blank',
};
