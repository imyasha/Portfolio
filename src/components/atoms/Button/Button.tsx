import { FC } from 'react';
import { StyledButton, StyledLink } from './styles';

export interface ButtonProps {
	href?: string;
	onClick?: (...args: any) => void;
	target?: '_blank' | '_self' | '_parent';
}

const Button: FC<ButtonProps> = ({ children, href, target, onClick }) => {
	if (href)
		return (
			<StyledLink href={href} target={target}>
				{children}
			</StyledLink>
		);

	return <StyledButton onClick={onClick}>{children}</StyledButton>;
};

export { Button };
