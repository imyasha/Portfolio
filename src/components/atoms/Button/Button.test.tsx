import { Button } from '../Button';
import { render } from '@testing-library/react';

describe('Button', () => {
	it('renders Button Link correctly', () => {
		const { container } = render(
			<Button href="/url" target="_blank">
				Button text
			</Button>
		);
		expect(container).toMatchSnapshot();
	});
	it('renders Button correctly', () => {
		const { container } = render(
			<Button onClick={jest.fn}>Button text</Button>
		);
		expect(container).toMatchSnapshot();
	});
});
