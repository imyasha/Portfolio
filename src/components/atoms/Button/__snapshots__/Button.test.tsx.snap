// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button Link correctly 1`] = `
<div>
  <a
    class="styles__StyledLink-a6av32-1 yMtKJ"
    href="/url"
    target="_blank"
  >
    Button text
  </a>
</div>
`;

exports[`Button renders Button correctly 1`] = `
<div>
  <button
    class="styles__StyledButton-a6av32-0 edzQTh"
  >
    Button text
  </button>
</div>
`;
